aws/rust-runtime/aws-config/src/profile/credentials/repr.rs

```diff
@@ -123,15 +122,11 @@ pub(super) fn resolve_chain<'a>(
     // - There is not default profile
     // Then:
     // - Treat this situation as if no profiles were defined
-    if profile_override.is_none()
-        && profile_set.selected_profile() == "default"
-        && profile_set.get_profile("default").is_none()
-    {
+    if profile_set.selected_profile() == "default" && profile_set.get_profile("default").is_none() {
         tracing::debug!("No default profile defined");
         return Err(ProfileFileError::NoProfilesDefined);
```

Comment:
Is it possible for a user to have other profiles but they forgot to set the profile name? If so, then I think this error could be improved. It's not so much that no profiles were defined but that the selected profile (`default`) couldn't be found.